docs/troubleshooting.md

```diff
@@ -0,0 +1,37 @@
+# Troubleshooting Tips
+
+When developing operators, there might occasionally be moments when Kubernetes
+APIs or apiserver don't return error message with full context or when the error
+message can be a bit misleading. Purpose of this document is to list some common
+pitfalls that might be hard to debug in some cases and where returned error
+message is not perfectly clear.
+
+## RBAC rules
+
+When `informer.Watch()` is called, it might fail with an "unknown error". This
+error is not very descriptive and there is work ongoing towards OperatorKit
+improving logging here as the original error is
+[StatusError](https://github.com/kubernetes/apimachinery/blob/master/pkg/api/errors/errors.go#L39)
+with `ErrStatus.Reason`:
+[Forbidden](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L569)
+Root cause here is a missing RBAC rule. The operator must have watch permission
+for the used CRD in order to be able to operate on it.
+
+```
+unknown (get clusternetworkconfigs.core.giantswarm.io)
+```
+
+
+## CRD registration
+
+When creating new CRD for an operator and having everything working, but getting
+an error from `streamwatcher.go` that states `unable to decode an event from the
+watch stream` when an object for corresponding CR is created, one possible
+mistake in this case is missing type registration when using generated CRD
+clientset. When type registration is missing, `client-go` decoder cannot
+recognize event and therefore fails with message like following:
```

Comment:
Maybe link to apiextensions where to register this things would be useful here - if this is about that.